### ISSUE TITLE
Blank statusDescription always appearing in customDimensions with Spring Boot 3.0.3-BETA

### ISSUE THREAD

**Issue by petergphillips** [microsoft/ApplicationInsights-Java#1555]
### Expected behavior
Custom dimensions that aren't set shouldn't appear in application insights.

### Actual behavior
Under 3.0.3-BETA we can now see
```
{"statusDescription":""}
```
always appearing in the `customDimensions` in `requests`.  This information was not present in 3.0.2.  Note that other custom dimensions appear correctly, it is just that we have an extra unnecessary one.

### To Reproduce
Wire up a Spring Boot 2.4.3 application with app insights and look in `requests`.

#### Sample Application
If applicable, provide a sample application which reproduces the issue.

### System information
Happy to provide a sample application if required.

SDK Version: 2.6.2
Java Agent Version: 3.0.3-BETA
OS type and version: Debian 14
Application Server type and version (if applicable): Spring Boot 2.4.3
Using spring-boot? yes

**Comment by petergphillips** [comment]
our applicationinsights.json:
```
{
  "role": {
    "name": "our-application-name"
  },
  "customDimensions": {
    "service.version": "${BUILD_NUMBER}"
  },
  "instrumentation": {
    "logging": {
      "level": "DEBUG"
    }
  },
  "selfDiagnostics": {
    "destination": "console"
  }
}
```

**Comment by petergphillips** [comment]
Note too that `service.version` doesn't appear in our `customDimensions`, since it is correctly transferred into `application_Version`.
This isn't a big issue for us, it is more to inform you that the `requests` behaviour has (I presume) accidentally changed since the last version.  None of our code breaks as a result, there is just some unnecessary extra information in `customDimensions`.

**Comment by trask** [comment]
Thanks @petergphillips, really appreciate the report! Will definitely fix in 3.0.3-BETA.2.
